packages/waku/src/cli.ts

```diff
@@ -96,6 +96,8 @@ async function runDev() {
   const config = await loadConfig();
   const app = new Hono();
   app.use('*', runner({ cmd: 'dev', config, env: process.env as any }));
+  // @ts-expect-error bypassing hono default notFound handler
+  app.use('*', (_c, _next) => {})
```

Comment:
@dai-shi I think I figured it out. The default hono notFound handler is not needed for the dev server. The `next()` call is running before the dev server sets the result and that invokes the not found handler here - https://github.com/honojs/hono/blob/main/src/hono-base.ts#L424

I added an additional middleware here to exit the middleware loop without invoking that final next(). I'm not sure if there's a better way to handle this.

Reply:
I should add that this is only a problem when the dev server is being used without the SSR middleware.

Reply:
Nice finding! I'm not sure either, but feels like there should be a better way. Maybe we find it in the future.